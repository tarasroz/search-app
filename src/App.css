
.ui-input {
  -webkit-box-shadow: 0 0.2rem 1rem rgba(0, 0, 0, 0.1);
          box-shadow: 0 0.2rem 1rem rgba(0, 0, 0, 0.1);
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
       appearance: none;
  width: 100%;
  border: none;
  font-size: 2rem;
  height: 3rem;
  margin-bottom: 1rem;
  padding: 0 1rem;
}
.ui-input::-webkit-input-placeholder {
  color: #CDCDCD;
}
.ui-input:focus {
  outline: none;
}

.ui-items {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -ms-flex-negative: 1;
      flex-shrink: 1;
  overflow-y: scroll;
  margin: 0 -.25rem;

}

.ui-item {
  display: block;
  height: 10rem;
  width: auto;
  -ms-flex-negative: 0;
      flex-shrink: 0;
  -webkit-box-flex: 0;
      -ms-flex-positive: 0;
          flex-grow: 0;
  margin: .25rem;
  -webkit-animation: item 0.5s calc(var(--i, 0) * .05s) cubic-bezier(0.5, 0, 0.2, 1) both;
          animation: item 0.5s calc(var(--i, 0) * .05s) cubic-bezier(0.5, 0, 0.2, 1) both;
  background-color: #EEE;
}

/*  */

.ui-app {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  width: 70rem;
  max-width: 90vw;
  height: calc(100vh - 2rem);
}

.ui-app:after {
  content: "current state: " attr(data-state);
  position: absolute;
  bottom: .5rem;
  color: white;
  background-color: rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  padding: .5rem 1rem;
  border-radius: 1rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
          transform: translateX(-50%);
  text-shadow: 0 0 .1rem black;
  pointer-events: none;
}

.ui-form {
  margin-bottom: 1rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column
}

.ui-button {
  -webkit-box-shadow: 0 0.2rem 1rem rgba(0, 0, 0, 0.1);
          box-shadow: 0 0.2rem 1rem rgba(0, 0, 0, 0.1);
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
       appearance: none;
  border: none;
  background-color: #EB7452;
  color: white;
  height: 3rem;
  padding: 0 3rem;
  border-radius: 3rem;
  margin: 0 1rem;
  outline: none  
}
.ui-button[disabled] {
  opacity: 0.5;
}
.ui-button[type="button"] {
  background-color: #555;
}

.ui-items:hover > .ui-item {
  opacity: 0.7;
}
.ui-items:hover > .ui-item:hover {
  opacity: 1;
}

@-webkit-keyframes item {
  from {
    -webkit-transform: scale(0);
            transform: scale(0);
  }
  to {
    -webkit-transform: scale(1);
            transform: scale(1);
  }
}
@keyframes item {
  from {
    -webkit-transform: scale(0);
            transform: scale(0);
  }
  to {
    -webkit-transform: scale(1);
            transform: scale(1);
  }
}

.ui-photo-detail {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.ui-photo {
  height: auto;
  width: auto;
  min-height: 50vh;
  min-width: 50vw;
  max-height: 80%;
  max-width: 80%;
} 
