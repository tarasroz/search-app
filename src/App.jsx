import React from "react";
import fetchJsonp from "fetch-jsonp";
import 'normalize.css'
import './App.css';

class App extends React.Component {

  state = {
    disableForm: false,
    query: "",
    pictures: []
  };

  enterLoading() {
    const encodedQuery = encodeURIComponent(this.state.query)

    setTimeout(() => {
      fetchJsonp(
        `https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=6b3575d10435de5f010fc941f5eff94a&per_page=15&tags=${encodedQuery}&format=json`,
        { jsonpCallback: "jsoncallback" }
      )
        .then(res => res.json())
        .then(j => {
          let picArray = j.photos.photo.map((pic, i) => {
            let srcPath = `https://farm${pic.farm}.staticflickr.com/${pic.server}/${pic.id}_${pic.secret}.jpg`;
            return (
              <div key={pic.id}>
                <img
                  className=".ui-photo"
                  alt="carts"
                  src={srcPath}
                  style={{ "--i": i }}
                />
                <p className=".ui-photo-detail">{pic.title}</p>
              </div>
            )
          })
          this.setState({ pictures: picArray });
        })
        .catch(e => console.log(e))
    }, 1000)
  };

  handleChangeQuery = e => {
    this.setState({ query: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.enterLoading();
  }

  render() {
    return (
      <div className="ui-app">
        <form className="ui-form" onSubmit={this.handleSubmit}>
          <input
            className="ui-input"
            disabled={this.state.disableForm}
            placeholder="Search Flickr for images..."
            type="search"
            value={this.state.query}
            onChange={this.handleChangeQuery}
          />
          <button
            className="ui-button"
            type="submit"
          >Search</button>
        </form>

        <section className="ui-items">
          {this.state.pictures}
        </section>
      </div>
    );
  }
}

export default App;
