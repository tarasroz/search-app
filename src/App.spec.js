import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App, { pictures } from './App'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App:', () => {

  it('renders children when passed in', () => {
    let wrapper = shallow(
      <section className="ui-items" />
    );
    expect(wrapper.contains(<section className="ui-items" />)).toEqual(true);
  });

  it('should set a list', () => {
    const state = { disableForm: false, query: "", pictures: [] };
    const newState = (state, {
      disableForm: false,
      query: "qwerty",
      pictures,
    });
    expect(newState).toEqual({ disableForm: false, query: "qwerty", pictures });
  });

  test('snapshot renders', () => {
    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('fetches async data', () => {
    const fetchJsonp = new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve({
          photos: {
            photo: [
              { title: 'a' },
              { title: 'b' },
            ],
          },
        })
      }, 1000)
    }
    );
    // let wrapper = mount(<App />);
    let wrapper = render(<App />);
    expect(wrapper.find('p')).toHaveLength(0);
    fetchJsonp.then(() => {
      expect(wrapper.find('p').length).toEqual(2);
    });
  });
})

