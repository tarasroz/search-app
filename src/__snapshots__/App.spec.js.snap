// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: snapshot renders 1`] = `
<div
  className="ui-app"
>
  <form
    className="ui-form"
    onSubmit={[Function]}
  >
    <input
      className="ui-input"
      disabled={false}
      onChange={[Function]}
      placeholder="Search Flickr for images..."
      type="search"
      value=""
    />
    <button
      className="ui-button"
      type="submit"
    >
      Search
    </button>
  </form>
  <section
    className="ui-items"
  />
</div>
`;
